apps/fortuna/src/command/run.rs

```diff
@@ -157,10 +161,23 @@ pub async fn run(opts: &RunOptions) -> Result<()> {
     ))?;
     let (tx_exit, rx_exit) = watch::channel(false);
 
-    let mut chains: HashMap<ChainId, BlockchainState> = HashMap::new();
+    let mut tasks = Vec::new();
     for (chain_id, chain_config) in &config.chains {
-        let state =
-            setup_chain_state(&config.provider.address, &secret, chain_id, chain_config).await;
+        tasks.push(
+            setup_chain_state(
```

Comment:
Example:
```rust
use std::time::Instant;
use futures::future::join_all;
use tokio::spawn;
pub async fn long_task() {
    let mut z:i32 = 3;
    for _ in 0..100000000 {
        z.overflowing_mul(z);
    }
}

#[tokio::main]
async fn main() {
    let current_time = Instant::now();
    let mut tasks = Vec::new();
    for _ in 0..5 {
        // tasks.push(long_task()); // This will run the tasks sequentially
        tasks.push(spawn(long_task())); // This will run the tasks concurrently
    }
    join_all(tasks).await;
    println!("Duration: {:?}", Instant::now().duration_since(current_time));
}
```